src/modules/fancyzones/editor/FancyZonesEditor/App.xaml.cs

```diff
@@ -27,46 +36,52 @@ public App()
             }
         }
 
-        private void OnStartup(object sender, StartupEventArgs e)
+        public void OnStartup(object sender, StartupEventArgs e)
         {
-            LayoutModel[] foundModel = new LayoutModel[NumMonitors];
+            FoundModel = new LayoutModel[NumMonitors];
 
             for (int setting = 0; setting < NumMonitors; setting++)
             {
                 foreach (LayoutModel model in ZoneSettings[setting].DefaultModels)
                 {
-                    if (model.Type == Settings.ActiveZoneSetLayoutType)
+                    if (model.Type == ZoneSettings[setting].ActiveZoneSetLayoutType)
                     {
                         // found match
-                        foundModel[setting] = model;
+                        FoundModel[setting] = model;
                         break;
                     }
                 }
 
-                if (foundModel == null)
+                if (FoundModel == null)

```

Comment:
Are you trying to check if you didn't find any active layout in the last loop? Then you will need to check FoundModel[setting] rather than FoundModel, which is never null.

Reply:
Absolutely right! Thanks 👌